code/parachain/frame/composable-maths/src/dex/constant_product.rs

```diff
@@ -490,6 +496,11 @@ pub fn compute_first_deposit_lp_<T: PerThing>(
 	Ok(ConstantProductAmmValueFeePair { value: product.safe_to_fixed_point()?, fee: 0 })
 }
 
+pub struct ConstantProductFirstDepositInput<T> {
```

Comment:
Let's not introduce new types to keep consistency for the moment with all interfaces/functions in this file. For now simply removing the `balance` parameter works for first deposit use as the balance is always 0 for first deposit.

Reply:
What consistency is being kept? Newtypes are used throughout this file (`ConstantProductAmmResult<_>`, `ConstantProductAmmValueFeePair`, `AssetAmount`, etc). I think using a newtype is actually _more_ consistent with what we currently have